iModelCore/iModelPlatform/DgnCore/linestyle/LsDb.cpp

```diff
@@ -631,9 +637,37 @@ DgnStyleId LineStyleElement::ImportLineStyle(DgnStyleId srcStyleId, DgnImportCon
         return DgnStyleId();
         }
 
-    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetAttributes(jsonObj));
+    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetUnitDef(jsonObj));
     return (result == BSISUCCESS) ? dstStyleId : DgnStyleId();
     }
 
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DgnElementPtr LineStyleElement::_CloneForImport(DgnDbStatus* status, DgnModelR destModel, DgnImportContext& context) const
+    {
+    DgnDbStatus _status;
+    if (nullptr == status)
+        status = &_status;
+
+
+    DgnElementPtr newElem = T_Super::_CloneForImport(status, destModel, context);
+    if (!newElem.IsValid() || (DgnDbStatus::Success != *status))
+        return newElem;
+
+    DgnStyleId destStyleId = context.RemapLineStyleId(DgnStyleId(this->m_elementId.GetValue()));

```

Comment:
This looks wrong to me? Should be getting data from "this", copying it to newElem and using the DgnImportContext to remap any ids it contains if context.IsBetweenDbs is true. Then you just return newElem. Don't think that call to Update should be here either...newElem will be inserted the first time this is called...and then never called again as a geometric element that references this style should find an existing id from the DgnImportContext.